migration-engine/connectors/sql-migration-connector/src/database_schema_calculator.rs

```diff
@@ -72,27 +73,49 @@ impl<'a> DatabaseSchemaCalculator<'a> {
                     sequence: None,
                 };
 
-                let indexes = model
-                    .fields()
-                    .filter_map(|f| {
-                        if f.is_unique {
-                            Some(Index {
-                                name: format!("{}.{}", &model.db_name(), &f.db_name()),
-                                columns: vec![f.db_name().clone()],
-                                tpe: IndexType::Unique,
-                            })
-                        } else {
-                            None
-                        }
-                    })
-                    .collect();
+                let single_field_indexes = model.fields().filter_map(|f| {
+                    if f.is_unique {
+                        Some(Index {
+                            name: format!("{}.{}", &model.db_name(), &f.db_name()),
+                            columns: vec![f.db_name().clone()],
+                            tpe: IndexType::Unique,
+                        })
+                    } else {
+                        None
+                    }
+                });
+
+                let multiple_field_indexes = model.indexes.iter().map(|index_definition: &IndexDefinition| Index {
+                    name: index_definition.name.clone().unwrap_or_else(|| {
+                        format!("{}.{}", &model.db_name(), index_definition.fields.iter().join("_"))
```

Comment:
Ideally we would use the database name of the fields instead of their name here. It is not too important though. Your call whether that is easy enough to do it.

Reply:
Changed in 514cc3a56497f3d6a94d011821959c15d33cfd52